### ISSUE TITLE
Port to windows and python2?

### ISSUE THREAD

**Issue by jeroen** [JanMarvin/v8-R#4]
I'd like to look into porting this package to Windows in: https://github.com/r-windows/rtools-packages

However one thing I noticed is that you use python2 which is currently being killed of everywhere. Is it possible to use python3 too?

**Comment by JanMarvin** [comment]
It's not yet possible to depend on python 3. Depot tools still requires python 2, but I'm sure that it'll be only a question of weeks

**Comment by jeroen** [comment]
OK thanks. Also, do you know if it is possible to build with gcc instead of clang? If you build this with clang on Arch, can R users link to the library if the R package is compiled with GCC?

**Comment by JanMarvin** [comment]
Yes, this is what I do. I only build it with clang because the gcc build was once broken. Not sure about the current status

**Comment by jeroen** [comment]
I tried to build it on windows but, as expected, I get stuck with python problems. Can you send a ping when depot_tools works on python3?

**Comment by JanMarvin** [comment]
Sure, will do

**Comment by JanMarvin** [comment]
python3 build is supported in https://github.com/JanMarvin/v8-R/tree/python3 tests still require python2.

**Comment by JanMarvin** [comment]
latest v8 builds with python3, tests still require python2

**Comment by jeroen** [comment]
Neat it would be cool if we can port it to msys2 and rtools!

**Comment by jeroen** [comment]
My ultimate goal is to update this one at some point: https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-v8

**Comment by JanMarvin** [comment]
I haven't played much attention to the Windows msys2 side of things, aside from your talk a few years ago in the Hague, but I do have a Windows installation running and could have a look (unless you're doing that anyways).

**Comment by JanMarvin** [comment]
Had a look. And [this](https://gist.github.com/JanMarvin/12df37c8d0f123ed1c88587ffa508d6d) modification of the v8-r PKGBUILD allows to checkout the source code using their python (presumably the easier part).
No clue how to tell v8 that the PKGBUILD wants a mingw build and that it is not supposed to look for any visual studio files. Most likely it will require some modifications to their BUILD.gn files.

**Comment by jeroen** [comment]
Somebody else also happens to been working on this right now: https://github.com/msys2/MINGW-packages/pull/6856. Can you have a quick look at his [PKGBUILD](https://github.com/Kwizatz/MINGW-packages/blob/v8/mingw-w64-v8/PKGBUILD) to see if it seems correct, from your experience?

**Comment by JanMarvin** [comment]
oh cool, sure i'll have a look. saves me some nerves digging further

**Comment by JanMarvin** [comment]
`v8_use_external_startup_data=true` iirc this did not work for us. Other that that I there are many changes to the build process I'm glad someone volunteered to do, but ideally they should be included upstream. Such long patches have a tendency to become annoying over time. My approach probably would be different, just copying a few build files into the build directory to avoid the hassle of modifying the patches once google modifies their files (and they will do so).
